src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueTrimmable.cs

```diff
@@ -51,5 +51,97 @@ public override void WriteTo(Utf8JsonWriter writer, JsonSerializerOptions? optio
                 JsonSerializer.Serialize(writer, _value, _jsonTypeInfo);
             }
         }
+
+        internal override JsonNode InternalDeepClone()
+        {
+            if (_converter is not null)
+            {
+                return _value is JsonElement element
+                    ? new JsonValueTrimmable<JsonElement>(element.Clone(), JsonMetadataServices.JsonElementConverter, Options)
+                    : new JsonValueTrimmable<TValue>(_value, _converter, Options);
+            }
+            else
+            {
+                Debug.Assert(_jsonTypeInfo != null);
+                return JsonSerializer.SerializeToNode(_value, _jsonTypeInfo)!;
+            }
+        }
+
+        internal override bool DeepEquals(JsonNode? node)
+        {
+            if (node is null)
+            {
+                return false;
+            }
+
+            if (_jsonTypeInfo is not null)
+            {
+                JsonNode? jsonNode = JsonSerializer.SerializeToNode(_value, _jsonTypeInfo);
+                return DeepEquals(jsonNode, node);
+            }
+            else
+            {
+                if (node is JsonArray || node is JsonObject)
+                {
+                    return false;
+                }
+
+                if (node is JsonValueTrimmable<JsonElement> jsonElementNodeOther && _value is JsonElement jsonElementCurrent)
+                {
+                    if (jsonElementNodeOther._value.ValueKind != jsonElementCurrent.ValueKind)
+                    {
+                        return false;
+                    }
+
+                    switch (jsonElementCurrent.ValueKind)
+                    {
+                        case JsonValueKind.String:
+                        case JsonValueKind.Number:
+                        case JsonValueKind.True:
+                        case JsonValueKind.False:
+                            return jsonElementCurrent.GetRawValue().Span.SequenceEqual(jsonElementNodeOther._value.GetRawValue().Span);
```

Comment:
This does not account for escaped strings when you have `JsonValueKind.String`

Reply:
At the same time you could call `GetBoolean()` for the case of `True`/`False`.

Reply:
thanks @eiriktsarpalis , I have updated it and added test case for escaped string case.

Reply:
> At the same time you could call `GetBoolean()` for the case of `True`/`False`.

Actually, wondering if just comparing with `jsonElementNodeOther.ValueKind` should suffice in this case:

```C#
                        case JsonValueKind.True:
                        case JsonValueKind.False:
                            return jsonElementCurrent.ValueKind == jsonElementNodeOther.ValueKind;
```